bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/discovery/ThingDiscoveryService.java

```diff
@@ -213,7 +214,8 @@ protected void addDevices(List<Device> devices, List<Room> rooms) {
     }
 
     protected String getRoomNameForDevice(Device device, List<Room> rooms) {
-        return rooms.stream().filter(room -> room.id.equals(device.roomId)).findAny().map(r -> r.name).orElse("");
+        return Objects.requireNonNull(
```

Comment:
I suppose this is necessary because the library cannot properly detect automatically that this is non-null? I would prefer to wait until the library does this properly instead of making the code more complicated than it has to be.

Reply:
Unfortunately the library is not expected to be updated with that enhancement anytime soon. Agree that there is room for improvement.
This PR aligns the dependency with the other repo’s and also brings some fixes